docs/language/learn-ql/writing-queries/debugging-queries.rst

```diff
@@ -0,0 +1,87 @@
+Query writing: common issues
+============================
+
+This topic offers some simple tips on how to avoid commons problems that can affect the performance of your QL queries.
+Before reading the tips below, it is worth reiterating a few important points about QL:
+
+- In QL :doc:`predicates <ql-handbook:predicates>` and :ref:`classes <ql-handbook:classes>` are all just database `relations <https://en.wikipedia.org/wiki/Relation_(database)>`__--that is, sets of tuples in a table. Large predicates generate tables with large numbers of rows, and are therefore expensive to compute.
+- QL is implemented using standard database operations and `relational algebra <https://en.wikipedia.org/wiki/Relational_algebra>`__ (such as join, projection, union, etc.). For further information about query languages and databases, see :doc:`About QL <about-ql>`.
+- QL is evaluated *bottom-up*, which means that a predicate is not evaluated until **all** of the predicates that it depends on are evaluated. For more information on query evaluation, see :doc:`Evaluation of QL programs <ql-handbook:evaluation>` in the QL handbook. 
+
+Performance tips
+================
+
+Follow the guidelines below to ensure that you don't get get tripped up by the most common QL performance pitfalls.
+
+Avoid cartesian products
```

Comment:
Not sure if "Avoid" is the right way to describe this.

This problem usually occurs accidentally, because the QL author forgot to add a necessary condition. It's usually also a correctness issue - the predicate will have more results than intended. I therefore think "avoid" is not the right way to describe it - perhaps "Eliminate cartesian products".